File: codingame/onboarding.py
#!/usr/bin/python
import sys

if sys.version_info[0]>=3: raw_input=input
while 1:
	enemy_1 = raw_input() # name of enemy 1
	dist_1 = int(raw_input()) # distance to enemy 1
	enemy_2 = raw_input() # name of enemy 2
	dist_2 = int(raw_input()) # distance to enemy 2
	print(enemy_1 if dist_1<dist_2 else enemy_2)

File: codingame/onboarding.dart
#!/usr/bin/env dart
import 'dart:io';
import 'dart:math';

void main() {
	for(;;){
		String enemy1 = stdin.readLineSync(); // name of enemy 1
		int dist1 = int.parse(stdin.readLineSync()); // distance to enemy 1
		String enemy2 = stdin.readLineSync(); // name of enemy 2
		int dist2 = int.parse(stdin.readLineSync()); // distance to enemy 2

		print(dist1<dist2 ? enemy1 : enemy2);
	}
}